.github/workflows/build-test-and-push.yml

```
@@ -0,0 +1,101 @@
+name: Build and Test Container
+
+on:
+  push:
+    branches: [ "**" ]
+  pull_request:
+    branches: [ "**" ]
+
+jobs:
+  build-test-and-push:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Build Docker image for testing
+      uses: docker/build-push-action@v5
+      with:
+        context: .
+        push: false
+        tags: test-image:latest
+        load: true
+
+    - name: Test container
+      run: |
+        # Start container in background with test environment variables
+        docker run -d --name test-container -p 8000:8000 \
```

Please always use the longhands in code for readability

```suggestion
        docker run --daemon --name test-container --port 8000:8000 \
```